akka-http-bench-jmh/src/main/scala/akka/http/impl/engine/http2/H2ServerProcessingBenchmark.scala

```diff
@@ -0,0 +1,95 @@
+package akka.http.impl.engine.http2
+
+import java.util.concurrent.CountDownLatch
+
+import scala.concurrent.Await
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import akka.actor.ActorSystem
+import akka.http.CommonBenchmark
+import akka.http.impl.engine.http2.FrameEvent.HeadersFrame
+import akka.http.impl.engine.http2.framing.FrameRenderer
+import akka.http.scaladsl.Http2
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.settings.ServerSettings
+import akka.stream.{ ActorMaterializer, OverflowStrategy }
+import akka.stream.TLSProtocol.{ SslTlsInbound, SslTlsOutbound }
+import akka.stream.scaladsl.{ Flow, Keep, Sink, Source }
+import akka.util.ByteString
+import com.typesafe.config.ConfigFactory
+import org.openjdk.jmh.annotations._
+
+class H2ServerProcessingBenchmark extends CommonBenchmark {
+  // Obtained by converting the input request bytes from curl with --http2-prior-knowledge
+  def request(streamId: Int) =
+    FrameRenderer.render(HeadersFrame(streamId, endStream = true, endHeaders = true, HPackSpecExamples.C41FirstRequestWithHuffman, None))
+
+  val response: HttpResponse = HPackSpecExamples.FirstResponse
+
+  var httpFlow: Flow[ByteString, ByteString, Any] = _
+  implicit var system: ActorSystem = _
+  implicit var mat: ActorMaterializer = _
+
+  val packedResponse = ByteString(-62, -63, -64, -65, -66)
+
+  @Benchmark
+  @OperationsPerInvocation(10000)
+  def benchRequestProcessing(): Unit = {
+    val numRequests = 10000
+    val latch = new CountDownLatch(numRequests)
+
+    val (in, done) =
+      Source.queue(numRequests, OverflowStrategy.fail)
```

Comment:
Could this just be a fixed source that concats the preface with a number of requests?